src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitService.kt

```diff
@@ -227,7 +227,7 @@ class VisitService(
 
   fun bookVisit(applicationReference: String): VisitDto {
 
-    val visitToBook = visitRepository.findByApplicationReference(applicationReference) ?: throw VisitNotFoundException("Could not find reserved visit applicationReference:$applicationReference not found")
+    val visitToBook = visitRepository.findApplication(applicationReference) ?: throw VisitNotFoundException("Application (reference $applicationReference) not found")
```

Comment:
Is the "Application (reference $applicationReference) not found" incorrect in this scenario as the visit might have been BOOKED? Should we instead check the visit status? Or set the message to not found or already booked?